doc/stages/filters.dbscan.rst

```diff
@@ -0,0 +1,45 @@
+.. _filters.dbscan:
+
+===============================================================================
+filters.dbscan
+===============================================================================
+
+The DBSCAN filter performs Density-Based Spatial Clustering of Applications
+with Noise (DBSCAN) [Ester1996]_ and labels each point with its associated
+cluster ID. Points that do not belong to a cluster are given a Cluster ID of
+-1. The remaining clusters are labeled as integers starting from 0.
+
+.. embed::
+
+Example
+-------
+
+.. code-block:: json
+
+  [
+      "input.las",
+      {
+          "type":"filters.dbscan",
+          "min_points":10,
+          "eps":2.0
+      },
+      {
+          "type":"writers.bpf",
+          "filename":"output.bpf",
+          "output_dims":"X,Y,Z,ClusterID"
+      }
+  ]
+
+Options
+-------
+
+min_points
```

Comment:
Number of dimensions?  Do you mean x, y and z?

Reply:
Yeah, so a little confusing if you are thinking about number of "PDAL Dimensions" = number of attributes in a point record. I do mean XYZ in this context, though in #2748 it has been suggested that clustering could in theory be performed on 1) more than just the three XYZ dimensions, or 2) a custom set of dimensions such as NxNyNz.

Reply:
Updated description slightly. The addition of the new `dimensions` argument will help to highlight the treatment of dimensions and number used.